components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -621,8 +664,17 @@ public void addResource(Resource resource) throws ConfigurationManagementExcepti
                             not create files. It is allowed to create a resource without files or attributes in order
                             to allow file upload after resource creation.
                             */
-                            preparedStatement.setBoolean(++initialParameterIndex, false);
-                            preparedStatement.setBoolean(++initialParameterIndex, isAttributeExists);
+                            try {
+                                if (isOracleDB() || isMSSqlDB()) {
+                                    preparedStatement.setInt(++initialParameterIndex, 0);
+                                    preparedStatement.setInt(++initialParameterIndex, isAttributeExists ? 1 : 0);
```

Comment:
as per https://github.com/wso2/carbon-identity-framework/pull/2847/files#diff-5b637a3090fa921e05d00ba7a47c9860R1494 even in DB2 we need to use 1 or 0 right? please check

Reply:
If "True" is given SMALLINT DEFAULT 0 datatype convert it to 1 and store. Similarly, 0 will be saved for "False". Thus, I didn't change insertion and update queries related to DB2.  Only  for the select queries, it should be handled. https://github.com/wso2/carbon-identity-framework/pull/2843/commits/5a0504e41b864eb30d7aa555f60249f9b40e79e9#r402250884